spec/acts_as_taggable_on/tagging_spec.rb

```diff
@@ -49,15 +49,74 @@
     ActsAsTaggableOn.remove_unused_tags = previous_setting
   end
 
-  pending 'context scopes' do
-    describe '.by_context'
-
-    describe '.by_contexts'
-
-    describe '.owned_by'
-
-    describe '.not_owned'
-
+  describe 'context scopes' do
+    before do
+      @tagging_2 = ActsAsTaggableOn::Tagging.new
+      @tagging_3 = ActsAsTaggableOn::Tagging.new
+
+      @tagger = User.new
+      @tagger_2 = User.new
+
+      @tagging.taggable = TaggableModel.create(name: "Black holes")
+      @tagging.tag = ActsAsTaggableOn::Tag.create(name: "Physics")
+      @tagging.tagger = @tagger
+      @tagging.context = 'Science'
+      @tagging.save
+
+      @tagging_2.taggable = TaggableModel.create(name: "Satellites")
+      @tagging_2.tag = ActsAsTaggableOn::Tag.create(name: "Technology")
+      @tagging_2.tagger = @tagger_2
+      @tagging_2.context = 'Science'
+      @tagging_2.save
+
+      @tagging_3.taggable = TaggableModel.create(name: "Satellites")
+      @tagging_3.tag = ActsAsTaggableOn::Tag.create(name: "Engineering")
+      @tagging_3.tagger = @tagger_2
+      @tagging_3.context = 'Astronomy'
+      @tagging_3.save
+
+    end
+
+    describe '.owned_by' do
+      it "should belong to a specific user" do
+        expect(@tagging).to be_valid
+        expect(@tagging_2).to be_valid
+
+        expect(ActsAsTaggableOn::Tagging.owned_by(@tagger).first).to eq(@tagging)
+        expect(ActsAsTaggableOn::Tagging.owned_by(@tagger_2).first).to eq(@tagging_2)
+      end
+    end
+
+    describe '.by_context' do
+      it "should be found by context" do
+        expect(ActsAsTaggableOn::Tagging.by_context('Science').length).to eq(2);
+      end
+    end
+
+    describe '.by_contexts' do
+      it "should find taggings by contexts" do
+        expect(@tagging_3).to be_valid
```

Comment:
Ditto... because here we're really testing our before block... once you're happy with it take these out I reckon